src/components/templates/PosterPage/components/PosterPageSettingsControl/PosterPageSettingsControl.tsx

```diff
@@ -0,0 +1,45 @@
+import React from "react";
+import { OverlayTrigger, Popover } from "react-bootstrap";
+import { faCog } from "@fortawesome/free-solid-svg-icons";
+
+import { Checkbox } from "components/atoms/Checkbox";
+
+import { PosterPageControl } from "../PosterPageControl";
+
+import "./PosterPageSettingsControl.scss";
+
+export interface PosterPageSettingsControlProps {
+  setVenueLiveOff: () => void;
+  setVenueLiveOn: () => void;
+  isPosterLive?: boolean;
+}
+
+export const PosterPageSettingsControl: React.FC<PosterPageSettingsControlProps> = ({
+  isPosterLive,
+  setVenueLiveOff,
+  setVenueLiveOn,
+}) => (
+  <OverlayTrigger
+    trigger="click"
+    placement="bottom-end"
+    overlay={
+      <Popover
```

Comment:
Done in https://github.com/sparkletown/sparkle/pull/1299/commits/8b6c3322f506c146fafa16a67707b7b74417aaf9

Reply:
Looking at the docs for `OverlayTrigger` it sounds like we're meant to be passing a ref to what we render?

- https://react-bootstrap.github.io/components/overlays/#overlay-trigger
  - > Note that triggering components must be able to accept a `ref` since `<OverlayTrigger>` will attempt to add one. You can use `forwardRef()` for function components.
- https://react-bootstrap.github.io/components/overlays/#customizing-trigger-behavior
  - > For more advanced behaviors `<OverlayTrigger>` accepts a function child that passes in the injected `ref` and event handlers that correspond to the configured `trigger` prop.

That still seems to be the case in the version we are using:

- https://github.com/react-bootstrap/react-bootstrap/blob/v1.0.1/www/src/pages/components/overlays.js
  - This still mentions it
- https://github.com/react-bootstrap/react-bootstrap/blob/v1.0.1/src/Overlay.js
- https://github.com/react-bootstrap/react-bootstrap/blob/v1.0.1/src/OverlayTrigger.js

Reply:
I played with passing refs for an hour and could do so that the popover is rendered below the button. I moved the Overlay component inside for the reason that I don't have the time to play with that anymore in https://github.com/sparkletown/sparkle/pull/1299/commits/4a604c64289e616d7ad1f36334b31dffaeb48f2e

Reply:
> I moved the Overlay component inside for the reason that I don't have the time/mental energy to play with that anymore

I don't understand why that is? The API docs/code imply that it still needs the `ref` regardless of whether it's extracted as a component passed in, or used inline like you reverted it to?

---

> I played with passing refs for an hour

Why? What was the issue? It should just literally be calling `forwardRef` like this:

```
export const PosterPageSettingsControlOverlay: React.FC<PosterPageSettingsControlProps> = forwardRef(
```

And then putting that passed ref onto the appropriate DOM element within it.

- https://reactjs.org/docs/forwarding-refs.html
- https://reactjs.org/docs/react-api.html#reactforwardref